tests/test_ctrlchannel

```diff
@@ -611,4 +611,27 @@ rm -f $LOG_FILE
 
 echo "OK"
 
+# Test CMD_SET_DATAFD
+cp ${PWD}/${DIR}/data/tpmstate1/* ${TPMDIR}
+$SWTPM_EXE socket --flags not-need-init \
+           --ctrl type=unixio,path=$SOCK_PATH \
+           --tpmstate dir=$TPMDIR -t --pid file=$PID_FILE \
+           --log file=$LOG_FILE,level=20 &>/dev/null &
+PID=$!
+
+if wait_for_file $PID_FILE 3; then
+	echo "Error: Socket TPM did not write pidfile."
+	exit 1
+fi
+
+LOG=$(SOCK_PATH=$SOCK_PATH exec python3 $DIR/test_setdatafd.py)
```

Comment:
I think this test program is missing. Also, can it not be written with functions available in python 2.7?

Reply:
Oops, ya i just realized that i missed that file.

And regarding python2.7, the issue is there is no way to send data + sockfd in a single stretch, we needs to send in two steps:
```
   socket.sendall(data)
  multiprocessing.sendfd(socket.fileno(), fds[1].fileno)
```
This not supported by swtpm server.

Reply:
I'll try to fix this today.

Reply:
by the way off topic, have you seen the Marc's comment on my qemu patchset about passing 'locality number' in big-endian. What do you suggest

Reply:
locality number is just a uint8_t. No big endian here.

Reply:
yes, i missed this point thanks.

Reply:
@avalluri I have this PR now https://github.com/stefanberger/swtpm/pull/40 for receiving control command data in chunks. Please have a look and comment. If it looks good to you, let me know and I'll merge.

Reply:
@avalluri : I created now branch 'amarnath' (https://github.com/stefanberger/swtpm/tree/amarnath) that applies (and rebases) your 2 patches from this PR on top of the 2 patches I have in PR #40. I added a patch on top of that to make things work with python 2.7. Can you clean up your 2nd patch and squash this fix-up patch?

I had to turn the order of sending the data and control messages around for python 2.7 in your python test program. This made it work. I am not sure why it works, though. In the original order it would send the control message first and swtpm's recvmsg() would return '1', which indicates that a single byte was received, but that is not possible. Subsequent waits for the data didn't work. The test case now works quite reliably with python 2.7 but it's certainly better to send the data in one chunk as python 3 seems to be capable of doing.